Explain env vs envFrom for page Define Environment Variables for a Container (#40055)
kubernetes/website: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md

that run in the Pod. To set environment variables, include the `env` or
`envFrom` field in the configuration file.

> Now lets look at the difference between `env` and `envFrom`

Reviewer: ```suggestion
The `env` and `envFrom` fields have different effects.
```

Author: suggested changes done.

> `env` : This is a field in the pod specification that allows you to set environment variables for a specific container.
> 
> `envFrom` :  This is a field in the pod specification that allows you to set environment variables by referencing a ConfigMap or a Secret. When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret are set as environment variables in the container. Hence `envFrom` is used to set multiple environment variables by referencing a ConfigMap or Secret

Reviewer: ```suggestion
`env`
: allows you to set environment variables for a container, specifying a value directly for each variable that you name.

`envFrom`
: allows you to set environment variables for a container by referencing either a ConfigMap or a Secret. When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret are set as environment variables for the container. You can also specify a common prefix string.

This page explains how to use `env`.
```

Author: suggested changes done.

> You can read more about <a href="https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables"> ConfigMap </a> and <a href="https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables"> Secret </a>

Reviewer: [Use Markdown, not HTML](https://kubernetes.io/docs/contribute/style/style-guide/#links), for links.

Author: suggested changes done.

that run in the Pod. To set environment variables, include the `env` or
`envFrom` field in the configuration file.

The `env` and `envFrom` fields have different effects.

`env`
: allows you to set environment variables for a container, specifying a value directly for each variable that you name.

`envFrom`

> : allows you to set environment variables for a container by referencing either a ConfigMap or a Secret. When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret are set as environment variables for the container. You can also specify a common prefix string.

Reviewer 2: ```suggestion
: allows you to set environment variables for a container by referencing either a ConfigMap or a Secret.
  When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret
  are set as environment variables for the container.
  You can also specify a common prefix string.
```

Author: suggested changes done.

> You can read more about [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables) and [Secret](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables)

Reviewer 2: ```suggestion
You can read more about [ConfigMap](/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables)
and [Secret](/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables).
```
